### PR TITLE
Fix bug that preventing switching back to ADX datasource

### PR DESCRIPTION
**Pull request by alyssabull**
This PR fixes a bug that threw an error when switching back to an ADX datasource from a different datasource.

### CODE REVIEW

**Review by aangelisc — CHANGES_REQUESTED** [review]
Because of overlap between the `KustoQuery` type and `ExpressionQuery` this change will cause the page to crash if an expression query is present and the user tries to switch to an ADX datasource. This is because of the addition of `query.expression` in the condition for rendering of the `VisualQueryEditor`. 

I think maybe we can just remove that requirement from the condition to make sure this still works as expected.

File: package.json [hunk 1]
```diff
@@ -87,6 +87,7 @@
     "@grafana/experimental": "^1.1.0",
     "@grafana/runtime": "9.2.1",
     "@grafana/ui": "9.2.1",
+    "@testing-library/dom": "^9.3.1",
```
Comment (asimpson) [inline package.json:90]: I'm not sure why this got added here? If you back out that change and re-run `yarn install --pure-lockfile` what happens?

**Review by asimpson — CHANGES_REQUESTED** [review]

**Review by aangelisc — APPROVED** [review]

File: yarn.lock [hunk 2]
```diff
@@ -2828,6 +2828,20 @@
     lz-string "^1.4.4"
     pretty-format "^26.6.2"
 
+"@testing-library/dom@^9.3.1":
```
Comment (asimpson) [inline yarn.lock:2831]: we  don't need these changes now since the package.json changes went away.

**Review by asimpson — CHANGES_REQUESTED** [review]

File: yarn.lock (hunk 2 above)
Comment (alyssabull) [reply yarn.lock]: Done - ty!

**Review by asimpson — APPROVED** [review]
Woot, nicely done, this one was tricky :trophy:

**Pull request merged**